fuel-txpool/src/service.rs

```diff
@@ -99,10 +161,13 @@ impl ServiceBuilder {
 
 pub struct Context {
     pub config: Config,
-    pub broadcast: broadcast::Sender<TxStatusBroadcast>,
     pub db: Arc<Box<dyn TxPoolDb>>,
-    pub receiver: mpsc::Receiver<TxPoolMpsc>,
-    pub import_block_events: broadcast::Receiver<ImportBlockBroadcast>,
+    pub txpool_receiver: mpsc::Receiver<TxPoolMpsc>,
+    pub tx_status_sender: broadcast::Sender<TxStatusBroadcast>,
+    pub import_block_receiver: broadcast::Receiver<ImportBlockBroadcast>,
+    pub incoming_tx_receiver: broadcast::Receiver<TransactionBroadcast>,
+    #[cfg(feature = "p2p")]
+    pub network_sender: mpsc::Sender<P2pRequestEvent>,
```

Comment:
Maybe we can already introduce the traits we discussed on the call in this PR?

You can represent P2P as `Gossiper` and `TransactionProvider` traits. It will aggregate `incoming_tx_receiver` and `network_sender` into one field, making the code cleaner. It seems easy to do because TXPool already knows nothing about  P2P(based on the imports in `Cargo.toml`).

<img width="273" alt="image" src="https://user-images.githubusercontent.com/18346821/193168847-c981aaa3-988a-4fdb-9d63-b7615bf41071.png">

Reply:
I'd prefer to checkpoint this PR as it's been in progress for a long time, and then we can do channel / trait conversion in subsequent PRs.